### PR TITLE
Bug 2024933: Jenkins fabric8 dependencies update, credentials sync test case 

### PR DESCRIPTION
**Pull request by alicerum**
In order to fix the problem with jenkins not syncing things on startup, `fabric8`'s `openshift-client` and `kubernetes-client` dependencies have been updated.

At the same time, `jenkins`'s version has been bumped due to new `kubernetes` plugin version, along with `jenkins`'s bom version.

This PR also contains test case which will check that credentials created before jenkins are correctly synced by it.

### CODE REVIEW

**Comment by alicerum** [comment]
/assign @akram 

/assign @jitendar-singh
for the qe

File: test/e2e/sync_plugin_test.go [hunk 1]
```diff
@@ -135,13 +135,15 @@ func setupThroughJenkinsLaunch(t *testing.T, ta *testArgs) *testArgs {
 	setupClients(ta.t)
 
 	randomTestNamespaceName := generateName(testNamespace)
-	ta.ns = randomTestNamespaceName
-	_, err := projectClient.ProjectV1().ProjectRequests().Create(context.Background(), &projectv1.ProjectRequest{
-		ObjectMeta: metav1.ObjectMeta{Name: randomTestNamespaceName},
-	}, metav1.CreateOptions{})
+	if len(ta.ns) == 0 {
```
Comment (gabemontero) [inline test/e2e/sync_plugin_test.go:138]: did you run `gofmt` @alicerum ?

if so, I've been running into the same  issue in this repo.  For some reason, `gofmt` works differently for golang files in this repo.

@adambkaplan @akram FYI

Comment (gabemontero) [reply test/e2e/sync_plugin_test.go:138]: aside from gofmt, move `randomTestNamespaceName := generateName(testNamespace)` inside the `if len(ta.ns) == 0` block, and switch to `ta.ns = generateName(testNamespaace)`

**Review by gabemontero — CHANGES_REQUESTED** [review]
one question and one minor suggestion

if you in fact did not run 
`gofmt -w ./test/`
I'd be curious if it produces changes for you.  If it does, add those with a separate commit.

File: test/e2e/sync_plugin_test.go (hunk 1 above)
Comment (alicerum) [reply test/e2e/sync_plugin_test.go:138]: `gofmt` does nothing for me here :/

Comment (adambkaplan) [reply test/e2e/sync_plugin_test.go:138]: I think we should address the formatting issue in a separate PR - seems to be a common issue (with perhaps the same root cause).

Comment (gabemontero) [reply test/e2e/sync_plugin_test.go:137]: Damn ... unfortunately:

```
gmontero ~/go/src/github.com/openshift/jenkins-sync-plugin  (add-sample-side-car-e2es)$ git rm .editorconfig 
rm '.editorconfig'
gmontero ~/go/src/github.com/openshift/jenkins-sync-plugin  (add-sample-side-car-e2es)$ gofmt -w ./test/
gmontero ~/go/src/github.com/openshift/jenkins-sync-plugin  (add-sample-side-car-e2es)$ git status 
On branch add-sample-side-car-e2es
Changes to be committed:
  (use "git restore --staged <file>..." to unstage)
	deleted:    .editorconfig

gmontero ~/go/src/github.com/openshift/jenkins-sync-plugin  (add-sample-side-car-e2es)$ 
```

Comment (gabemontero) [reply test/e2e/sync_plugin_test.go:137]: though maybe it renders in github with the right spacing? ... I noticed in my goland I have 4 space indentation now, even though gofmt does not "change" the file

Comment (gabemontero) [reply test/e2e/sync_plugin_test.go:137]: but so it does not get lost in the shuffle @alicerum still make the change I noted at https://github.com/openshift/jenkins-sync-plugin/pull/408#discussion_r766768340 

orthogonal to gofmt

Comment (alicerum) [reply test/e2e/sync_plugin_test.go:137]: @gabemontero already pushed the changed

as for editor, it only shows github how to show tabs, @adambkaplan and me checked that actual go files contain tab identation and everything is alright :D

Comment (gabemontero) [reply test/e2e/sync_plugin_test.go:137]: ok thanks for confirming my "though maybe it renders..." speculation

I double confirmed when pushing a commit to #406 which removes .editorconfig ... the tabbing looks better in that PR

**Comment by gabemontero** [comment]
You've got a compile failure in the test @alicerum 

```
KUBERNETES_CONFIG=/tmp/kubeconfig-3377180997 go test -timeout 75m -v ./test/e2e/...
# github.com/openshift/jenkins-sync-plugin/test/e2e [github.com/openshift/jenkins-sync-plugin/test/e2e.test]
test/e2e/sync_plugin_test.go:140:40: undefined: randomTestNamespaceName
FAIL	github.com/openshift/jenkins-sync-plugin/test/e2e [build failed]
FAIL
make: *** [test-e2e] Error 2
```

**Comment by alicerum** [comment]
@gabemontero ouch, thank you

**Comment by gabemontero** [comment]
terraform fail

/retest

**Comment by gabemontero** [comment]
OK @akram @adambkaplan @alicerum I attempted to cut version v1.0.52 of the sync plugin, but either something changed upstream or I somehow lost my registered maven credentials / login, and the push failed:

```
    [INFO] --- maven-deploy-plugin:2.8.2:deploy (default-deploy) @ openshift-sync ---
    Uploading to maven.jenkins-ci.org: https://repo.jenkins-ci.org/releases/io/fabric8/jenkins/plugins/openshift-sync/1.0.51/openshift-sync-1.0.51.hpi
    Uploading to maven.jenkins-ci.org: https://repo.jenkins-ci.org/releases/io/fabric8/jenkins/plugins/openshift-sync/1.0.51/openshift-sync-1.0.51.pom
    [INFO] ------------------------------------------------------------------------
    [INFO] BUILD FAILURE
    [INFO] ------------------------------------------------------------------------
    [INFO] Total time:  22.922 s
    [INFO] Finished at: 2021-12-10T15:51:27-05:00
    [INFO] ------------------------------------------------------------------------
    [ERROR] Failed to execute goal org.apache.maven.plugins:maven-deploy-plugin:2.8.2:deploy (default-deploy) on project openshift-sync: Failed to deploy artifacts: Could not transfer artifact io.fabric8.jenkins.plugins:openshift-sync:hpi:1.0.51 from/to maven.jenkins-ci.org (https://repo.jenkins-ci.org/releases/): authentication failed for https://repo.jenkins-ci.org/releases/io/fabric8/jenkins/plugins/openshift-sync/1.0.51/openshift-sync-1.0.51.hpi, status: 401 Unauthorized -> [Help 1]
    [ERROR] 
    [ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
    [ERROR] Re-run Maven using the -X switch to enable full debug logging.
    [ERROR] 
    [ERROR] For more information about the errors and possible solutions, please read the following articles:
    [ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/MojoExecutionException
```

Short term, I'll undo the new release commits and tags.

And when @akram next logs in, he can generate the new version v1.0.51.

I'll then spend time on Monday digging up the old links I had on getting permissions / research via internet search and talking with @akram how to re-establish my local settings so I can regain this capability for next time.

I'll report back here after I've reset the commits and tags in https://github.com/jenkinsci/openshift-sync-plugin/

**Comment by gabemontero** [comment]
OK the v1.0.51 commits and tags are removed.  I have already pushed the post 1.0.50 commits from openshift/jenkins-sync-plugin to jenkinsci/openshift-sync-plugin

Should be ready @akram for you to run `mvn release:prepare release:perform`

**Comment by akram** [comment]
@gabemontero starting that then.
I will let you know if I have something wrong.

**Comment by akram** [comment]
@gabemontero accounts.jenkins.io returned back to normal and so I pushed the release. And 1.0.51 is now available.

**Pull request merged**